File: apps/s-2-admin/src/api/schedule/ScheduleCreateInput.ts
import { UserWhereUniqueInput } from "../user/UserWhereUniqueInput";
import { EventTypeCreateNestedManyWithoutSchedulesInput } from "./EventTypeCreateNestedManyWithoutSchedulesInput";
import { AvailabilityCreateNestedManyWithoutSchedulesInput } from "./AvailabilityCreateNestedManyWithoutSchedulesInput";

export type ScheduleCreateInput = {
  user: UserWhereUniqueInput;
  name: string;
  timeZone?: string | null;
  eventType?: EventTypeCreateNestedManyWithoutSchedulesInput;
  availability?: AvailabilityCreateNestedManyWithoutSchedulesInput;
};

File: apps/cactus-demo/src/APIs/Schedule/Dtos/ScheduleCreateInput.cs
namespace CactusDemo.APIs.Dtos;

public class ScheduleCreateInput
{
    public int? Id { get; set; }

    public string Name { get; set; }

    public string? TimeZone { get; set; }

    public User Users { get; set; }

    public List<Availability>? Availability { get; set; }

    public List<EventType>? EventType { get; set; }
}
